clientv3/lease.go

```diff
@@ -184,6 +190,40 @@ func (l *lessor) Close() error {
 	return nil
 }
 
+func (l *lessor) keepAliveCtxCloser(ctx context.Context, id lease.LeaseID) {
+	activeCtx := true
+	for activeCtx {
+		select {
+		case <-ctx.Done():
+			activeCtx = false
```

Comment:
`activeCtx == false` => close the channel if it's still in the keepAlive for the id
`<-l.donec` => channel close is handled by recv shutdown (I guess it's OK to break out of the loop instead of returning as well, but it'd still need that activeCtx var)